src/lowering/desugar.ml

```diff
@@ -68,6 +68,22 @@ and exp' at note = function
     I.PrimE (I.RelPrim (!ot, o), [exp e1; exp e2])
   | S.ShowE (ot, e) ->
     I.PrimE (I.ShowPrim !ot, [exp e])
+  | S.ToCandidE (ot, es) ->
+    begin match T.normalize (!ot) with
+    | T.Tup ts1 ->
+      (optE (primE (I.SerializePrim ts1) (exps es))).it
+    | _ -> assert false
+    end
+  | S.FromCandidE (ot, e) ->
+    (* begin match T.normalize note.Note.typ with *) (* not a T.Tup _ *)
+    begin match T.normalize (!ot) with
+    | T.Opt t ->
+      begin match T.normalize t with
+        | T.Tup ts1 -> I.PrimE (I.DeserializePrim ts1, [exp e])
+        | t (* XXX check me. *) -> I.PrimE (I.DeserializePrim [t], [exp e])
```

Comment:
@crusso I needed line just above `(XXX check me)` to get the test `idl.mo` to pass without triggering an assertion failure here.